### PR TITLE
haskellPackages.pandoc-crossref: fix build and more

### PR DESCRIPTION
**Pull request by rnhmjoj**
###### Motivation for this change
pandoc-crossref managed to break in both 20.03 and unstable but I need it badly.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox))
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested by building a pandoc document
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/haskell-modules/configuration-common.nix [hunk 1]
```diff
@@ -1193,10 +1192,9 @@ self: super: {
   });
 
   # Remove unecessary constraint:
-  # https://github.com/agrafix/superbuffer/pull/2
-  superbuffer = overrideCabal super.superbuffer (drv: {
+  data-accessor-template = overrideCabal super.data-accessor-template (drv: {
     postPatch = ''
-      sed -i 's#QuickCheck < 2.10#QuickCheck < 2.13#' superbuffer.cabal
+      sed -i 's#template-haskell >=2.11 && <2.15#template-haskell#' data-accessor-template.cabal
```
Comment (cdepillabout) [inline pkgs/development/haskell-modules/configuration-common.nix]: Could you add a comment on this line explaining why it is needed, and when we can expect to remove it?  And ideally report this upstream and link to the issue here?

**Review by cdepillabout — CHANGES_REQUESTED** [review]
Aside from the one comment above, this looks good.

Feel free to go ahead and merge in when the comment above has been addressed.

File: pkgs/development/haskell-modules/configuration-common.nix (hunk 1 above)
Comment (rnhmjoj) [reply pkgs/development/haskell-modules/configuration-common.nix]: Done, thank you.

**Comment by rnhmjoj** [comment]
@cdepillabout Do you have any information about the LTS update in 20.03? I delayed a fix for pandoc-crossref in 20.03 because the upcoming update (and this PR at this point) would have fixed it.

**Comment by cdepillabout** [comment]
@rnhmjoj At one point, I was under the impression that @peti was going to do the update to LTS-15 in 20.03, but then I feel like I saw something saying he decided not to do it.

@peti, are you still planning on updating 20.03 to LTS-15, or is that no longer planned?

**Pull request merged**